### ISSUE TITLE
FASTRCNN example (mAP) at around 0.6937

### ISSUE THREAD

**Issue by pmfcdb** [Microsoft/CNTK#939]
Hello,

I got the example FASTRCCN running, but the (mAP) is at  0.6937, and the (Epoch 30) errs= 1.552%. I changed from 17 epochs to 30 to see if the result would improve, but the mAP is almost the same.

I posted a video in youtube recording every steps, you can see everything, from python version, to pip list, script execution. 

https://www.youtube.com/watch?v=_NSqAo82ySU

10:47 - Epoch 30 errs= 1.552%
11:20 (mAP) at around 0.6937

What can I do to improve mAP to 0.86?

Thks,

**Comment by PatrickBue** [comment]
Hi,
My guess is that the difference in mAP you are seeing is due to the test set being super small (it's just a toy example), and the training is non-deterministic. What's your mAP on the training set? It should be much higher (for that just run A3 again but with image_set = 'train'

**Comment by pmfcdb** [comment]
Hi,
My mAP with image_set to train is 0.8791.  How can I continue to improve the model? Can I change the PretrainedModels  AlexNet.model with the generated from training in ( FastRCNN\proc\grocery_2000\cntkFiles\Output ) and continue to improve?
Thks,

**Comment by pkranen** [comment]
Can you try using the [pre-trained model](https://www.cntk.ai/Models/FRCN_Grocery/Fast-RCNN.model) to see whether you get 0.86 mAP on the test set? Some details are provided at [https://github.com/Microsoft/CNTK/wiki/Object-Detection-using-Fast-R-CNN#using-a-pre-trained-model](https://github.com/Microsoft/CNTK/wiki/Object-Detection-using-Fast-R-CNN#using-a-pre-trained-model). Thanks.

**Comment by pmfcdb** [comment]
I used your pre-trained model but something went wrong because I only achieved Mean AP = 0.1182. 

I tested several times, generating my model from the beginning and then using your pre-trained model. So maybe the problem is on my cntk compilation? 
Thks,

**Comment by cha-zhang** [comment]
I verified that with the pre-trained model you should see 0.8688 mAP. Could you pull from the master and rebuild?

**Comment by pmfcdb** [comment]
Problem solved, I was using python 3.4.4 ( https://www.python.org/downloads/release/python-344/ )  and know I switch for Anaconda (Anaconda3-2.2.0-Windows-x86_64 -python3.4) and I'm using CNTK-2-0-beta1-0-Windows-64bit-GPU.

Now I have :
Evaluating detections
AP for         avocado = 1.0000
AP for          orange = 0.2500
AP for          butter = 1.0000
AP for       champagne = 1.0000
AP for          eggBox = 0.7500
AP for          gerkin = 1.0000
AP for         joghurt = 1.0000
AP for         ketchup = 1.0000
AP for     orangeJuice = 1.0000
AP for           onion = 1.0000
AP for          pepper = 0.7600
AP for          tomato = 0.6400
AP for           water = 0.5000
AP for            milk = 1.0000
AP for         tabasco = 1.0000
AP for         mustard = 1.0000
Mean AP = 0.8688

**Comment by mogrysama** [comment]
I have an equal problem. I always get a mAP of 0.5032. After setting image_set = 'train' I get 0.4875. I´m using CNTK for Windows v.2.0 Beta 6 CPU only and did everything like it is wrtitten here. Can anybody help?
